tasks/deploy/config/sepolia/dai-14day.ts

```diff
@@ -1,121 +1,119 @@
-                tx = await baseToken.write.setPublicCapability([
-                    toFunctionSelector("mint(address,uint256)"),
-                    true,
-                ]);
-                await pc.waitForTransactionReceipt({ hash: tx });
-                // approve the coordinator for the contribution
-                tx = await baseToken.write.approve([
-                    hre.hyperdriveDeploy.deployments.byName(
-                        "ERC4626_COORDINATOR",
-                    ).address,
-                    parseEther(CONTRIBUTION),
-                ]);
-                await pc.waitForTransactionReceipt({ hash: tx });
-                tx = await baseToken.write.mint([parseEther(CONTRIBUTION)]);
-                await pc.waitForTransactionReceipt({ hash: tx });
-            },
-        },
-        vaultSharesToken: {
-            name: "SDAI",
-            deploy: async (hre) => {
-                let pc = await hre.viem.getPublicClient();
-                let baseToken =
-                    hre.hyperdriveDeploy.deployments.byName("DAI").address;
-                let vaultSharesToken =
-                    await hre.hyperdriveDeploy.deployContract(
-                        "SDAI",
-                        "MockERC4626",
-                        [
-                            baseToken,
-                            "Savings DAI",
-                            "SDAI",
-                            parseEther("0.13"),
-                            "0xd94a3A0BfC798b98a700a785D5C610E8a2d5DBD8",
-                            true,
-                            parseEther("10000"),
-                        ],
-                    );
-                // allow minting by the public
-                let tx = await vaultSharesToken.write.setPublicCapability([
-                    toFunctionSelector("mint(uint256)"),
-                    true,
-                ]);
-                await pc.waitForTransactionReceipt({ hash: tx });
-                tx = await vaultSharesToken.write.setPublicCapability([
-                    toFunctionSelector("mint(address,uint256)"),
-                    true,
-                ]);
-                await pc.waitForTransactionReceipt({ hash: tx });
+    poolDeployConfig: async (hre) => {
+        return {
+            baseToken: hre.hyperdriveDeploy.deployments.byName("DAI").address,
+            vaultSharesToken:
+                hre.hyperdriveDeploy.deployments.byName("SDAI").address,
+            circuitBreakerDelta: parseEther("0.6"),
+            minimumShareReserves: parseEther("10"),
+            minimumTransactionAmount: parseEther("0.001"),
+            positionDuration: parseDuration("14 days"),
+            checkpointDuration: parseDuration("1 day"),
+            timeStretch: 0n,
+            governance: "0xc187a246Ee5A4Fe4395a8f6C0f9F2AA3A5a06e9b",
+            feeCollector: "0xc187a246Ee5A4Fe4395a8f6C0f9F2AA3A5a06e9b",
+            sweepCollector: "0xc187a246Ee5A4Fe4395a8f6C0f9F2AA3A5a06e9b",
```

Comment:
Would be good to read these from the factory. The deployment will fail if the details are different from the factory details.

Reply:
Totally agree with you.

Unfortunately doing this is be pretty non-trivial since we get the `PoolDeployConfig` type directly from the abi, so we'd have to start dissecting it a good bit.

Nice middle ground might be auto-filling them when we do config generation from the env file... will keep it in mind